### PR TITLE
activitywatch: add module

### PR DESCRIPTION
**Pull request by foo-dogsquared**
### Description

Adding [ActivityWatch](https://activitywatch.net/) service which is [fully intended to be local for the user](https://docs.activitywatch.net/en/latest/remote-server.html) so adding this to home-manager is a nice fit.

This service is fully deployed and managed by systemd instead of the [traditional way of using it by running `aw-qt`](https://docs.activitywatch.net/en/latest/getting-started.html#usage) which opens up a system tray.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

### CODE REVIEW

**Comment by foo-dogsquared** [comment]
Updated the module, resolved the merge conflicts, and rebased it on top of 6a8444467c83c961e2f5ff64fb4f422e303c98d3.

**Comment by foo-dogsquared** [comment]
Rebased the PR to current master revision and also updated the tests and the news item.

**Comment by voobscout** [comment]
Is this going to be merged anytime soon?

**Comment by rycee** [comment]
Seems I accidentally made review comments instead of real comments so they never showed up 🙁

By the way, there is no need to add a news item, we can add it when doing the merge. To help avoid these merge conflicts.

File: modules/services/activitywatch.nix [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+        ]
+      '';
+    };
+
+    watchers = lib.mkOption {
+      description = ''
+        Watchers to be included with the service alongside with their
+        configuration.
+
+        If configuration is set, a file will be generated in
```
[the remark this answers is not in the document]
Comment (foo-dogsquared) [reply modules/services/activitywatch.nix:171]: Done.

**Comment by foo-dogsquared** [comment]
All right, I've pushed with the suggested changes and added a new test.

> By the way, there is no need to add a news item, we can add it when doing the merge. To help avoid these merge conflicts.

Got it.

**Comment by rycee** [comment]
Thanks! I made a few minor final fixes, mainly to reduce the download size for the tests. Merged to master now 🙂

**Pull request merged**